web/massastation/src/pages/store/MyPlugin.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from 'react';
+
+import {
+  Button,
+  Certificate,
+  MassaWallet,
+  Plugin,
+} from '@massalabs/react-ui-kit';
+import { FiArrowUpRight, FiRefreshCcw, FiTrash2 } from 'react-icons/fi';
+import { IMassaPlugin } from './MyStation';
+
+enum PluginStatus {
+  Up = 'Up',
+  Down = 'Down',
+}
+
+function MyPlugin({ plugin }: { plugin: IMassaPlugin }) {
+  const [myPlugin, setMyPlugin] = useState<IMassaPlugin>(plugin);
+  function handlePluginState(method: string) {
+    // TODO :
+    // const {mutate } = usePost<any>("plugin-manager");
+    // const ok = mutate({method:"${method}"})
+    const newStatus = method === 'start' ? PluginStatus.Up : PluginStatus.Down;
+    const updatedPlugin = { ...myPlugin, status: newStatus };
+    setMyPlugin(updatedPlugin);
+  }
+  const argsOn = {
+    preIcon:
+      myPlugin.author === 'MassaLabs' ? (
+        <MassaWallet variant="rounded" />
+      ) : (
+        <img src={myPlugin.logo} />
+      ),
+    topAction: (
+      <Button onClick={() => handlePluginState('stop')} variant="toggle">
+        on
+      </Button>
+    ),
+    title: `${myPlugin.name} `,
```

Comment:
myPlugin on line 39, then on line 40, then on line 44, etc...

Isn't possible to extract that and use only the variable without this concatenation!? 🤔 

`let {a, b, c, d... } = myPlugin`

Reply:
would this be updated after a setMyPlugin ?

Reply:
yes !